src/auth/src/credentials/external_account_sources.rs

```diff
@@ -0,0 +1,15 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+pub mod url_sourced_account;
```

Comment:
we could have url sourced cred as a top level cred instead of keeping it inside another mod. The use statement is going to become pretty long for our users if we keep doing sub modules.

Reply:
are users going to reference the external account provider directly ? I thought users would use the `ExternalAccountConfig` struct to ask for a External Account credential type, and then we figured out internally which type of implementation and source users need. That's how it works in Go ([see here](https://github.com/googleapis/google-cloud-go/blob/main/auth/credentials/internal/externalaccount/externalaccount.go#L235). I'll check other implementations

Reply:
Customers could try building creds from config, or for whatever reason, they may choose to build through code. So we should make it easy to initialize in both ways.

See https://github.com/googleapis/google-cloud-rust/blob/ae91098c611434626e9809a1cab470ab81b870a0/src/auth/src/credentials/user_account.rs#L131 for ref